### PR TITLE
Addressing complexity comments from Static Analysis

### PR DESCRIPTION
**Pull request by acabarbaye**

### CODE REVIEW

File: mbed_tools_ci_scripts/utils/git_helpers.py [hunk 1]
```diff
@@ -580,6 +580,20 @@ def uncommitted_changes(self) -> List[Path]:
                 line.strip().split(' ')[-1]) for line in status.splitlines()
         ]
 
+    @staticmethod
+    def _applying_modifications(destination: Path,
```
Comment (muchzill4) [inline mbed_tools_ci_scripts/utils/git_helpers.py]: `applying_*` is a very odd name for a method.

File: mbed_tools_ci_scripts/utils/filesystem_helpers.py [hunk 2]
```diff
@@ -119,14 +119,17 @@ def __exit__(self, exc: Any, value: Any, tb: Any) -> None:
         """Context manager exit point."""
         self.cleanup()
 
+    def _cleanup_using_lib_facility(self) -> None:
+        try:
+            self._tmp_dir_context_manager.cleanup()
+        except FileNotFoundError as e:
```
Comment (muchzill4) [inline mbed_tools_ci_scripts/utils/filesystem_helpers.py:125]: Is this necessary?

File: mbed_tools_ci_scripts/utils/filesystem_helpers.py [hunk 3]
```diff
@@ -119,14 +119,17 @@ def __exit__(self, exc: Any, value: Any, tb: Any) -> None:
...
         """Deletes the temporary directory."""
-        if not (platform.system() == 'Windows'):
-            try:
-                self._tmp_dir_context_manager.cleanup()
-            except FileNotFoundError as e:
-                logger.warning(
-                    f"Failed cleaning up {self.path}. Reason: {str(e)}")
-        else:
+        if platform.system() == 'Windows':
             if self.path.exists():
```
Comment (muchzill4) [inline mbed_tools_ci_scripts/utils/filesystem_helpers.py]: Pretty sure using `ignore_errors` makes this conditional obsolete.

File: mbed_tools_ci_scripts/utils/filesystem_helpers.py [hunk 4]
```diff
@@ -119,14 +119,17 @@ def __exit__(self, exc: Any, value: Any, tb: Any) -> None:
...
         self.cleanup()
 
+    def _cleanup_using_lib_facility(self) -> None:
+        try:
+            self._tmp_dir_context_manager.cleanup()
+        except FileNotFoundError as e:
+            logger.warning(
+                f"Failed cleaning up {self.path}. Reason: {str(e)}")
+
     def cleanup(self) -> None:
```
Comment (muchzill4) [inline mbed_tools_ci_scripts/utils/filesystem_helpers.py:133]: I find it odd that in non-windows case this calls `self.something_else()` and in windows case it does the work here. My gut feeling is this should either do all the work:
```
if platform.system() == 'Windows':
    shutil.rmtree(str(self.path), ignore_errors=True)
else:
    self._tmp_dir_context_manager.cleanup()
```

Or, if you insist on doing `try/except`, delegate always:

```
if platform.system() == 'Windows':
    self._windows_cleanup()
else:
    self._regular_cleanup() # needs a better name
```

**Review by muchzill4 — CHANGES_REQUESTED** [review]
+27 -14, sick! 👍

File: news/20200204.misc [hunk 5]
```diff
@@ -0,0 +1 @@
+Addressing some static analysis comments regarding complexity
```
Comment (mark-edgeworth) [inline news/20200204.misc:1]: Nit: missing newline

Comment (acabarbaye) [reply news/20200204.misc:1]: There are no newlines in news files

File: mbed_tools_ci_scripts/utils/git_helpers.py (hunk 1 above)
Comment (acabarbaye) [reply mbed_tools_ci_scripts/utils/git_helpers.py]: any suggestion?

Comment (muchzill4) [reply mbed_tools_ci_scripts/utils/git_helpers.py]: `apply_*`

File: mbed_tools_ci_scripts/utils/filesystem_helpers.py (hunk 3 above)
Comment (acabarbaye) [reply mbed_tools_ci_scripts/utils/filesystem_helpers.py]: it is to avoid getting things in the logs for no real reasons

Comment (muchzill4) [reply mbed_tools_ci_scripts/utils/filesystem_helpers.py]: What would be an example?

File: mbed_tools_ci_scripts/utils/filesystem_helpers.py (hunk 4 above)
Comment (acabarbaye) [reply mbed_tools_ci_scripts/utils/filesystem_helpers.py:133]: done

File: mbed_tools_ci_scripts/utils/filesystem_helpers.py (hunk 2 above)
Comment (acabarbaye) [reply mbed_tools_ci_scripts/utils/filesystem_helpers.py:125]: yes. cleanup may be used multiple times (ensure idempotency)

**Review by muchzill4 — APPROVED** [review]

**Pull request merged**